### PR TITLE
Migrate netstandard1.6 to netcoreapp2.0, update NuGetProvider submodule

### PR DESCRIPTION
**Pull request by brywang-msft**
Upgrade of all packages. The most recent System.Management.Automation package we can use for full CLR is 6.0.0-alpha17, as net451 support is removed afterwards. If we need to take a fix from that package, we'll need a different way to get the dll.

### CODE REVIEW

File: src/Microsoft.PackageManagement.MsuProvider/Microsoft.PackageManagement.MsuProvider.csproj [hunk 1]
```diff
@@ -16,22 +16,22 @@
...
-  </ItemGroup>
-
+  </ItemGroup>
+
   <ItemGroup>
-    <PackageReference Include="System.Management.Automation" Version="1.0.0-alpha9" />
-  </ItemGroup>
-
+    <PackageReference Include="System.Management.Automation" Version="6.0.0-alpha17" />
+  </ItemGroup>
```
Comment (jianyunt) [inline src/Microsoft.PackageManagement.MsuProvider/Microsoft.PackageManagement.MsuProvider.csproj]: If I remember correctly, we on purposely selected the first version of PowerShell Core to compile OneGet against. The reason is there is no changes in PSCOre or full powershell in this version so we can ensure the OneGet assemblies work on downlevel OSs. Please double check. thx

Comment (brywang-msft) [inline src/Microsoft.PackageManagement.MsuProvider/Microsoft.PackageManagement.MsuProvider.csproj]: PS4 seems fine after some small testing, but I can revert this after the NuGetProvider build is fixed.

File: appveyor.yml [hunk 2]
```diff
@@ -51,4 +51,6 @@ deploy_script:
 
 test_script:
  - ps: >-
-     & "c:\projects\oneget\Test\run-tests.ps1"
\ No newline at end of file
+     & "c:\projects\oneget\Test\run-tests.ps1"
+on_finish:
+  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
```
Comment (jianyunt) [inline appveyor.yml]: Bryan, any reasons we need to add this line at the end? I noticed you added it in line 13 already.

Comment (brywang-msft) [inline appveyor.yml]: Temporarily blocking for RDP to debug tests. Will remove when tests are working.

**Pull request merged**